ostd/src/task/processor.rs

```diff
@@ -90,6 +90,8 @@ pub(super) fn switch_to_task(next_task: Arc<Task>) {
     //    context and the next task context.
     // 2. The next task context is a valid task context.
     unsafe {
+        (*next_task_ctx_ptr).activate_tls_pointer();
+
         // This function may not return, for example, when the current task exits. So make sure
         // that all variables on the stack can be forgotten without causing resource leakage.
         context_switch(current_task_ctx_ptr, next_task_ctx_ptr);
```

Comment:
This is a little bit tricky. Since from the activation of the TLS pointer to the context switch, neither preemption nor IRQs are disabled. So if on SMP preemptive cases the task may be moved to another CPU before the switch happen. Should we `cli` at some place above and call `sti` after loading the registers in the context switch assemby?

Reply:
Or maybe activate the TLS pointer in the assembly along with the code loading the registers. Which may be better.

Reply:
> activate the TLS pointer in the assembly along with the code loading the registers

Done.